mkopn/installer.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+
+# disable terminal receiver
+stty -cread
+
+# enable tracing
+set -eux
+
+# sanity check
+[ ${1} = /dev/mmcblk0p3 ] || exit 1
+[ ${2} = /dev/mmcblk0p4 ] || exit 1
+
+IMAGE_FILE=${3}
+
+# write root
+ROOT_IMAGE_OFFSET=$(printf "%d" "0x$(hexdump -s136 -n8 -v -e "8/1 \"%02x\"" ${IMAGE_FILE})")
+dd if=${IMAGE_FILE} bs=1024 skip=$((${ROOT_IMAGE_OFFSET} / 1024)) | gunzip -c | dd of=/dev/mmcblk0p3
```

Comment:
Yes, because the root partition is recreated and its start is placed at the same location as before. I need to do it in this order to prevent an extra copy of the root filesystem because I may wipe the partition (data partition may change its layout pretty surely) the root filesystem is lying on.